test/hotspot/jtreg/runtime/Monitor/GuaranteedAsyncDeflationIntervalTest.java

```diff
@@ -0,0 +1,161 @@
+ * @library /test/lib
+ * @run driver GuaranteedAsyncDeflationIntervalTest
+ */
+
+public class GuaranteedAsyncDeflationIntervalTest {
+
+    public static void main(String[] args) throws Exception {
+        if (args.length == 0) {
+          driver();
+        } else {
+          test();
+        }
+    }
+
+    // Inflate a lot of monitors, so that threshold heuristics definitely fires
+    public static final int MONITORS = 10_000;
+
+    public static Object[] monitors;
+
+    public static void test() throws Exception {
+        monitors = new Object[MONITORS];
+        for (int i = 0; i < MONITORS; i++) {
+            Object o = new Object();
+            synchronized (o) {
+                try {
+                    o.wait(1); // Inflate!
+                } catch (InterruptedException ie) {
+                }
+            }
+            monitors[i] = o;
+        }
+
+        try {
+            Thread.sleep(10_000);
+        } catch (InterruptedException ie) {
+        }
+    }
+
+    public static void driver() throws Exception {
+        final String MSG_THRESHOLD  = "Async deflation needed: monitors used are above the threshold";
+        final String MSG_GUARANTEED = "Async deflation needed: guaranteed interval reached";
+
+        // Try with all heuristics disabled
+        {
+            ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
+                "-Xmx100M",
+                "-XX:+UnlockDiagnosticVMOptions",
+                "-XX:GuaranteedAsyncDeflationInterval=0",
+                "-XX:MonitorUsedDeflationThreshold=0",
+                "-Xlog:monitorinflation=info",
+                "GuaranteedAsyncDeflationIntervalTest",
+                "test");
+
+            OutputAnalyzer oa = new OutputAnalyzer(pb.start());
+            oa.shouldHaveExitValue(0);
+
+            oa.shouldNotContain(MSG_THRESHOLD);
+            oa.shouldNotContain(MSG_GUARANTEED);
+            assertNoDeflations(oa);
+        }
+
+        // Try with guaranteed interval only enabled
+        {
+            ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
+                "-Xmx100M",
+                "-XX:+UnlockDiagnosticVMOptions",
+                "-XX:GuaranteedAsyncDeflationInterval=100",
+                "-XX:MonitorUsedDeflationThreshold=0",
+                "-Xlog:monitorinflation=info",
+                "GuaranteedAsyncDeflationIntervalTest",
+                "test");
+
+            OutputAnalyzer oa = new OutputAnalyzer(pb.start());
+            oa.shouldHaveExitValue(0);
+
+            oa.shouldNotContain(MSG_THRESHOLD);
+            oa.shouldContain(MSG_GUARANTEED);
+            assertDeflations(oa);
+        }
+
+        // Try with both threshold heuristics and guaranteed interval enabled
+        {
+            ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
+                "-Xmx100M",
+                "-XX:+UnlockDiagnosticVMOptions",
+                "-XX:GuaranteedAsyncDeflationInterval=5000",
```

Comment:
Why not setting AsyncDeflationInterval to a very high value, effectively disabling it, and then check that we don't see MSG_THRESHOLD but still see MSG_GUARANTEED?

Reply:
`MUDT=0` has the similar effect. But yes, we can also test `ADI=0` here, added.